### PR TITLE
Migrate release 1.2.x distribution from jenkins to github actions

### PR DESCRIPTION
**Pull request by NipunaMadhushan**
## Purpose
Currently 1.2.x distribution has been publishing by using a jenkins job. With this effort, we can get rid of jenkins job for ballerina-distribution release by creating a prerelease in `ballerina-distribution` repository and archieve artifacts there.

## Goals
To get rid of dependencies external dependencies (eg: jenkins jobs need a physical machine inside the office to run) of our release pipeline.

## FIxes
https://github.com/ballerina-platform/ballerina-distribution/issues/2841

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    name: Publish Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '11'
```
Comment (keizer619) [inline .github/workflows/publish-release.yml:17]: @NipunaMadhushan Since 1.2.x support JDK8 lets we use that to build it?

File: pom.xml [hunk 2]
```diff
@@ -58,7 +58,7 @@
     <scm>
         <url>https://github.com/ballerina-platform/ballerina-distribution.git</url>
         <developerConnection>scm:git:https://github.com/ballerina-platform/ballerina-distribution.git</developerConnection>
-        <connection>scm:git:https://github.com/ballerina-platform/ballerina-distribution.git</connection>
+        <connection>scm:git:https://github.com/NipunaMadhushan/ballerina-distribution.git</connection>
```
Comment (keizer619) [inline pom.xml:61]: @NipunaMadhushan This has to be restored to `ballerina-platform`

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (NipunaMadhushan) [reply .github/workflows/publish-release.yml:17]: @keizer619 sure.

File: pom.xml (hunk 2 above)
Comment (NipunaMadhushan) [reply pom.xml]: @keizer619 addressed the suggestions.

**Review by keizer619 — APPROVED** [review]

**Pull request merged**